web/src/main/webapp/dataViews/heavyHittersView.ts

```diff
@@ -119,43 +165,65 @@ export class HeavyHittersView extends RemoteTableObjectView {
                 row.push(textToDiv(this.valueToString(tdv.rows[i].count)));
                 row.push(textToDiv(this.valueToString((tdv.rows[i].count / tdv.rowCount) * 100)));
                 row.push(new DataRange(position, tdv.rows[i].count, tdv.rowCount).getDOMRepresentation());
-                table.addElementRow(row);
+                let tRow : HTMLTableRowElement = this.table.addElementRow(row);
+                tRow.oncontextmenu = e => this.createAndShowContextMenu(tRow, e);
                 position += tdv.rows[i].count;
             }
+            if (this.restPos == tdv.rows.length) {
+                this.showRest(tdv.rows.length, position, this.restCount, tdv.rowCount, this.table);
+            }
         }
-        table.addFooter();
-        this.topLevel.appendChild(table.getHTMLRepresentation());
+        this.table.addFooter();
         this.page.scrollIntoView();
         this.page.reportTime(elapsedMs);
     }
 
-    private getRestCount(tdv:NextKList): number{
-        if (tdv.rows == null)
-            return tdv.rowCount;
-        else {
-            let runCount = tdv.rowCount;
-            for (let i = 0; i < tdv.rows.length; i++)
-                runCount -= tdv.rows[i].count;
-            return runCount;
-        }
-    }
 
-    private getRestPos(tdv:NextKList, restCount: number): number{
-        if (tdv.rows == null)
-            return 0;
-        else {
-            let i = 0;
-            while((i < tdv.rows.length) && (restCount <= tdv.rows[i].count))
-                i++;
-            return i;
+    private createAndShowContextMenu(tRow: HTMLTableRowElement,  e: MouseEvent): void {
+        if (e.ctrlKey && (e.button == 1)) {
```

Comment:
This pattern shows up too many times, we should move it to ui.ts or util.ts in a separate function.